File: backend-api/src/main/java/com/example/usermanagement/entity/MealRecord.java
package com.example.usermanagement.entity;

import javax.persistence.*;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 餐食记录实体
 */
@Entity
@Table(name = "meal_records")
public class MealRecord {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "meal_id")
    private Integer mealId;
    
    @Column(name = "user_id", nullable = false)
    private Long userId;
    
    @Column(name = "meal_date", nullable = false)
    private LocalDate mealDate;
    
    @Column(name = "meal_time", nullable = false)
    private LocalDateTime mealTime;
    
    @Column(name = "meal_type", nullable = false, length = 20)
    @Enumerated(EnumType.STRING)
    private MealType mealType;
    
    @Column(name = "drink_id")
    private Integer drinkId;
    
    @Column(name = "food_name", length = 200)
    private String foodName;
    
    @Column(name = "portion_size")
    private Float portionSize;
    
    @Column(name = "calories")
    private Float calories;
    
    @Column(name = "sugar_content")
    private Float sugarContent;
    
    @Column(name = "image_path", length = 500)
    private String imagePath;
    
    @Column(name = "notes", columnDefinition = "TEXT")
    private String notes;
    
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
    
    // Constructors
    public MealRecord() {}
    
    public MealRecord(Long userId, LocalDate mealDate, LocalDateTime mealTime, 
                     MealType mealType, String foodName, Float calories, Float sugarContent) {
        this.userId = userId;
        this.mealDate = mealDate;
        this.mealTime = mealTime;
        this.mealType = mealType;
        this.foodName = foodName;
        this.calories = calories;
        this.sugarContent = sugarContent;
    }
    
    // Getters and Setters
    public Integer getMealId() {
        return mealId;
    }
    
    public void setMealId(Integer mealId) {
        this.mealId = mealId;
    }
    
    public Long getUserId() {
        return userId;
    }
    
    public void setUserId(Long userId) {
        this.userId = userId;
    }
    
    public LocalDate getMealDate() {
        return mealDate;
    }
    
    public void setMealDate(LocalDate mealDate) {
        this.mealDate = mealDate;
    }
    
    public LocalDateTime getMealTime() {
        return mealTime;
    }
    
    public void setMealTime(LocalDateTime mealTime) {
        this.mealTime = mealTime;
    }
    
    public MealType getMealType() {
        return mealType;
    }
    
    public void setMealType(MealType mealType) {
        this.mealType = mealType;
    }
    
    public Integer getDrinkId() {
        return drinkId;
    }
    
    public void setDrinkId(Integer drinkId) {
        this.drinkId = drinkId;
    }
    
    public String getFoodName() {
        return foodName;
    }
    
    public void setFoodName(String foodName) {
        this.foodName = foodName;
    }
    
    public Float getPortionSize() {
        return portionSize;
    }
    
    public void setPortionSize(Float portionSize) {
        this.portionSize = portionSize;
    }
    
    public Float getCalories() {
        return calories;
    }
    
    public void setCalories(Float calories) {
        this.calories = calories;
    }
    
    public Float getSugarContent() {
        return sugarContent;
    }
    
    public void setSugarContent(Float sugarContent) {
        this.sugarContent = sugarContent;
    }
    
    public String getImagePath() {
        return imagePath;
    }
    
    public void setImagePath(String imagePath) {
        this.imagePath = imagePath;
    }
    
    public String getNotes() {
        return notes;
    }
    
    public void setNotes(String notes) {
        this.notes = notes;
    }
    
    public LocalDateTime getCreatedAt() {
        return createdAt;
    }
    
    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }
    
    /**
     * 餐次类型枚举
     */
    public enum MealType {
        breakfast, lunch, dinner, snack
    }
}

File: app/src/main/java/com/example/myapplication/model/MealRecord.kt
package com.example.myapplication.model

import com.google.gson.annotations.SerializedName
import java.util.Date

/**
 * 饮食日记记录
 */
data class MealRecord(
    @SerializedName("meal_id")
    val mealId: Int? = null,
    
    @SerializedName("user_id")
    val userId: Int,
    
    @SerializedName("meal_date")
    val mealDate: String, // 格式: yyyy-MM-dd
    
    @SerializedName("meal_time")
    val mealTime: String, // 格式: HH:mm:ss
    
    @SerializedName("meal_type")
    val mealType: String, // breakfast, lunch, dinner, snack
    
    @SerializedName("drink_id")
    val drinkId: Int? = null,
    
    @SerializedName("food_name")
    val foodName: String,
    
    @SerializedName("image_path")
    val foodImagePath: String? = null,
    
    @SerializedName("sugar_content")
    val sugarContent: Double, // 克
    
    @SerializedName("calories")
    val calories: Double, // 卡路里
    
    @SerializedName("protein")
    val protein: Double? = null, // 蛋白质(克)
    
    @SerializedName("fat")
    val fat: Double? = null, // 脂肪(克)
    
    @SerializedName("carbohydrate")
    val carbohydrate: Double? = null, // 碳水化合物(克)
    
    @SerializedName("portion_size")
    val portionSize: String? = null, // 份量描述
    
    @SerializedName("notes")
    val notes: String? = null, // 备注
    
    @SerializedName("created_at")
    val createdAt: String? = null
)

/**
 * 每日饮食记录汇总
 */
data class DailyMealSummary(
    @SerializedName("date")
    val date: String,
    
    @SerializedName("meals")
    val meals: List<MealRecord>,
    
    @SerializedName("total_sugar")
    val totalSugar: Double,
    
    @SerializedName("total_calories")
    val totalCalories: Double,
    
    @SerializedName("meal_count")
    val mealCount: Int
)

/**
 * 添加饮食记录请求
 */
data class AddMealRequest(
    @SerializedName("user_id")
    val userId: Int,
    
    @SerializedName("meal_date")
    val mealDate: String,
    
    @SerializedName("meal_time")
    val mealTime: String,
    
    @SerializedName("meal_type")
    val mealType: String,
    
    @SerializedName("food_name")
    val foodName: String,
    
    @SerializedName("sugar_content")
    val sugarContent: Double,
    
    @SerializedName("calories")
    val calories: Double,
    
    @SerializedName("protein")
    val protein: Double? = null,
    
    @SerializedName("fat")
    val fat: Double? = null,
    
    @SerializedName("carbohydrate")
    val carbohydrate: Double? = null,
    
    @SerializedName("portion_size")
    val portionSize: String? = null,
    
    @SerializedName("notes")
    val notes: String? = null,
    
    @SerializedName("image_path")
    val imagePath: String? = null
)

/**
 * 餐次类型
 */
enum class MealType(val value: String, val displayName: String) {
    BREAKFAST("breakfast", "早餐"),
    LUNCH("lunch", "午餐"),
    DINNER("dinner", "晚餐"),
    SNACK("snack", "加餐");
    
    companion object {
        fun fromValue(value: String): MealType {
            return values().find { it.value == value } ?: SNACK
        }
        
        fun getCurrentMealType(): MealType {
            val hour = java.util.Calendar.getInstance().get(java.util.Calendar.HOUR_OF_DAY)
            return when {
                hour in 6..10 -> BREAKFAST
                hour in 11..13 -> LUNCH
                hour in 17..20 -> DINNER
                else -> SNACK
            }
        }
    }
}

